dask_match/core.py

```diff
@@ -454,6 +462,61 @@ def _layer(self):
         }
 
 
+class ProjectPartitions(Expr):
```

Comment:
The term `Project` may be misused here.  

My understanding is that `Projection` is a term in relational algebra that is a fancy way of saying "select some columns".  I don't think that it gets applied to rows in general. 

Maybe `FilterPartitions` would be better.

Reply:
Or at least more in line with the nomenclature of that field.